yii-taskforce/frontend/controllers/TasksController.php

```diff
@@ -88,16 +104,162 @@ public function actionCreate()
+        }
+    }
+
+    /**
+     * Сохраняет отклик на задание
+     *
+     * @return \yii\web\Response
+     */
+    public function actionRespond()
+    {
+        $post = Yii::$app->request->post();
+
+        if(!empty($post)) {
+            throw new NotFoundHttpException('Страница не найдена');
+        }
+
+        $model = new Replies();
+        if($model->load($post) && $model->validate()) {
+            $model->user_id = Yii::$app->user->identity->id;
+            $model->save();
+        }
+
+        return $this->redirect(['tasks/view', 'id' => $post['Replies']['task_id']]);
+    }
+
+    /**
+     * Отказ от задания исполнителем
+     *
+     * @return \yii\web\Response
+     */
+    public function actionRefuse()
+    {
+        $post = Yii::$app->request->post();
+        if(!empty($post) && !empty($post['task_id'])) {
+            throw new NotFoundHttpException('Страница не найдена');
+        }
+
+        $model = Tasks::findOne(['id' => $post['task_id']]);
+        if(!$model) {
+            throw new NotFoundHttpException('Страница не найдена');
+        }
+
+        $actions = $model->getAvailableActions();
+
+        if(in_array(Tasks::ACTION_REFUSE, $actions)) {
+            $model->updateAttributes(['status' => Tasks::STATUS_FAILED]);
+        }
+
+        return $this->redirect(['tasks/view', 'id' => $post['task_id']]);
+    }
+
+    /**
+     * Отмена задания
+     *
+     * @return \yii\web\Response
+     */
+    public function actionCancel()
+    {
+        $post = Yii::$app->request->post();
+
+        if(!empty($post) && !empty($post['task_id'])) {
+            throw new NotFoundHttpException('Страница не найдена');
+        }
+
+        $model = Tasks::findOne(['id' => $post['task_id']]);
+        if(!$model) {
+            throw new NotFoundHttpException('Страница не найдена');
+        }
+
+        $actions = $model->getAvailableActions();
+
+        if(in_array(Tasks::ACTION_CANCEL, $actions)) {
+            $model->updateAttributes(['status' => Tasks::STATUS_CANCELED]);
+        }
+
+        return $this->redirect(['tasks/view', 'id' => $post['task_id']]);
+    }
+
+    /**
+     * Завершение задания и отзыв на исполнителя
+     *
+     * @return \yii\web\Response
+     */
+    public function actionDone()
+    {
+        $post = Yii::$app->request->post();
+        if(!empty($post)) {
+            throw new NotFoundHttpException('Страница не найдена');
+        }
+
+        $modelTask = Tasks::findOne(['id' => $post['Opinions']['task_id']]);
+        if(!$modelTask) {
+            throw new NotFoundHttpException('Страница не найдена');
+        }
+
+        $actions = $modelTask->getAvailableActions();
+
+        if(in_array(Tasks::ACTION_DONE, $actions)) {
+
+            $transaction = Yii::$app->db->beginTransaction();
+
+            $modelTask->updateAttributes(['status' => $post['completion'] == 'y' ? Tasks::STATUS_COMPLETED : Tasks::STATUS_FAILED]);
+
+            $model = new Opinions();
```

Comment:
```suggestion
            $option = new Opinions();
```
Вообще не очень хорошо, что сущности во множественном числе. У тебя ж тут одна опция в сущности, а не несколько.